File: c/adding_numbers.c
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>

int main(void) {
  int amount = 0;
  printf("Numbers to add? ");
  scanf("%d", &amount);

  int numbers[amount];
  int total = 0;

  for (int i = 0; i < amount; i++) {
    printf("Enter a number: ");
    scanf("%d", &numbers[i]);

    if (!isdigit(numbers[i])) {
      total += numbers[i];
    }
  }

  printf("The total is %d\n", total);

  return EXIT_SUCCESS;
}

File: zig/adding_numbers.zig
const std = @import("std");

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    var buffer: [1024]u8 = undefined;

    std.debug.print("How many numbers to add? ", .{});
    var input = try stdin.readUntilDelimiter(buffer[0..], '\n');

    const numbers_limit: u32 = try std.fmt.parseInt(u32, input, 10) catch 0;
    var sum: i32 = 0;

    for (0..numbers_limit) |_| {
        std.debug.print("Enter a number: ", .{});
        input = try stdin.readUntilDelimiter(buffer[0..], '\n');

        const value = std.fmt.parseInt(i32, input, 10) catch 0;
        sum += value;
    }

    std.debug.print("The total is: {d}", .{sum});
}
